src/algebra/parity.lean

```diff
@@ -241,6 +241,21 @@ lemma odd.neg_one_pow (h : odd n) : (-1 : α) ^ n = -1 := by rw [h.neg_pow, one_
 
 end monoid
 
+section canonically_ordered_comm_semiring
+
+variables [canonically_ordered_comm_semiring α]
+
+-- TODO: this holds more generally in a `canonically_ordered_add_monoid` if we refactor `odd` to use
```

Comment:
Would this be desirable, though? I thought the current definition of `odd` was a compromise - weaker assumptions could be made, but they'd be awkward and really badly behaved.

Reply:
I mean, it's worth noting regardless. I'm happy to demote it from a todo though.

Reply:
```suggestion
-- this holds more generally in a `canonically_ordered_add_monoid` if we refactor `odd` to use
```